src/classlibnative/bcltype/system.cpp

```diff
@@ -358,6 +358,13 @@ INT32 QCALLTYPE SystemNative::GetProcessorCount()
         processorCount = CPUGroupInfo::GetNumActiveProcessors();
     }
 
+#ifdef FEATURE_PAL
+    size_t cpuLimit;
+
+    if (PAL_GetCpuLimit(&cpuLimit))
+       processorCount = cpuLimit;
```

Comment:
The limit should only limit the count, but it should not be able to provide a larger value than the real count of processors if someone sets the period and quota so that their ratio is larger than the real count of processors.